Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Orientable Manifolds
+
+This file defines orientable manifolds: a differentiable manifold is orientable if and only if it
+admits an orientable atlas, i.e. an atlas whose transition functions have a strictly positive
+Jacobian.
+
+## Main Definitions
+
+- `OrientationPreserving` : a map between normed spaces is orientation-preserving on a given set
+  if the determinant of its Jacobian is strictly positive on that set.
+- `OrientationReversing` : a map between normed spaces is orientation-reversing on a given set
+  if the determinant of its Jacobian is strictly negative on that set.
+- `orientationPreservingGroupoid` : the groupoid of partial homeos of `H` which are
+  orientation-preserving.
+- `OrientableManifold` : a type class saying that the charted space `M`, modelled on the space
+  `H`, admits an orientation.
+- `OrientableSmoothManifold` : a type class representing a manifold that is both orientable
+  and smooth.
+
+## Main Results
+
```

Comment:
What do you think about mentioning that 0-dimensional manifolds are always orientable?